src/chapters/more-on-functions/recursion-walkthrough2.rst

```diff
@@ -0,0 +1,133 @@
+      }
+   }
+
+Now we are ready to take the next step.
+
+A Visual Representation
+------------------------
+
+For arrays with more than one entry, we need to make ``combineEntries``
+consider the first element and then *check what is left in the array*. If the
+rest of the array contains more than one item, ``combineEntries`` calls iteslf
+again and repeats the process with a smaller set of entries.
+
+To help visualize what's going on, let's start with the base case ``['L']``:
+
+.. figure:: figures/base-case-recursion.png
+   :alt: Visual representation for the base case.
+
+Nothing complicated here.  ``combineEntries`` sees only one item in the array,
+so it returns ``'L'``.
+
+Now consider an array with two elements ``['L', 'C']``:
+
+.. figure:: figures/second-case-recursion.png
+   :alt: Visual representation for the second-easiest case.
+
+In this case, ``combineEntries`` executes the ``else`` statement. We have no
+code for this yet, but we can still consider the logic:
+
+a. ``combineEntries`` returns ``'L'`` and calls itself again using what is left
+   inside the array (``['C']``).
+b. When passed ``['C']``, which is the base case, ``combineEntries`` returns
+   ``'C'``.
+c. The strings ``'L'`` and ``'C'`` get combined and returned as the final
+   result.
+
+Next, consider an array with three elements ``['L', 'C', '1']``:
+
+.. figure:: figures/third-case-recursion.png
+   :alt: Visual representation for the third-easiest case.
+
+As before, ``combineEntries`` executes the ``else`` statement, and we can
+follow the logic:
+
+a. ``combineEntries`` returns ``'L'`` and calls itself again using what is left
+   inside the array (``['C', '1']``).
+b. When passed ``['C', '1']``, ``combineEntries`` returns ``'C'`` and calls
+   itself again using what is left inside the array (``['1']``).
+c. When passed ``['1']``, which is the base case, ``combineEntries`` returns
+   ``'1'``.
+d. The strings ``'C'`` and ``'1'`` get combined and returned.
+e. The strings ``'L'`` and ``'C1'`` get combined and returned as the final
+   result.
+
+As we make the array longer, ``combineEntries`` calls itself more times. Each
+call evaluates a smaller and smaller section of the array until reaching the
+base case. This sets up a series of return events - each one selecting a
+single entry from the array. Rather than building ``'LC101'`` from left to
+right, recursion constructs the string starting with the base case and
+adding new characters to the front:
+
+| ``'1'``
+| ``'01'``
+| ``'101'``
+| ``'C101'``
+| ``'LC101'``
+
+A Function Calls Itself
+------------------------
+
+So how do we code the ``else`` statement in ``combineEntries``? Recall what
+needs to happen each time the statement runs:
+
+a. Select the first element in the array,
+b. Call ``combineEntries`` again with a smaller array.
+
+Bracket notation takes care of part a: ``arrayName[0]``.
+
+| For part b, remember that the :ref:`slice method <slice-examples>` returns
```

Comment:
It seems that these manual breaks are being use to solve issues that should be handled with CSS (not enough padding between paragraphs) or an RST directive (wanting a code sample to display on a new line). 

For a style issue, let's document it in Trello (with file name and screenshot) and leave the markup as-is (without the fix). 

For the second part (if displaying `arr = ...` on a separate line is indeed your intention), use `::` or ``.. sourcecode:: js``.

Reply:
It was the second part.  Splitting the code over two lines looked bad in this case.

Fixed.